### PR TITLE
multistage trainer draft

### PR DESCRIPTION
**Pull request by theRealSuperMario**
Implements #156

### CODE REVIEW

File: edflow/iterators/tf_trainer.py [hunk 1]
```diff
@@ -316,3 +316,100 @@ def run(self, fetches, feed_dict):
...
+            if end < 0:
+                return stage_number
+            elif global_step < end:
+                return stage_number
+            else:
+                pass
+
+
+class TFMultiStageModel(object):
+    def define_graph(self):
```
Comment (theRealSuperMario) [inline edflow/iterators/tf_trainer.py]: I am not sure if it is really useful to override `define_graph`. Maybe `setup_stages` is a better idea. The child class will have to call the parent anyway.

Comment (pesser) [reply edflow/iterators/tf_trainer.py]: why not just put it into `__init__`?

File: edflow/iterators/tf_trainer.py [hunk 2]
```diff
@@ -316,3 +316,100 @@ def run(self, fetches, feed_dict):
...
+        self._stage_update = tf.placeholder(tf.int32, shape=(), name="stage_update")
+        self._stage = tf.Variable(2, trainable=False, dtype=tf.int32)
```
Comment (pesser) [inline edflow/iterators/tf_trainer.py]: init to 0 is more intuitive, no?

File: examples/multistage_trainer/data.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import tensorflow as tf
```
Comment (pesser) [inline examples/multistage_trainer/data.py:1]: can we use `edflow.datasets.mnist.MNIST` instead?

File: examples/multistage_trainer/nn.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import tensorflow as tf
```
Comment (pesser) [inline examples/multistage_trainer/nn.py]: use `nn` included in edflow?

File: examples/multistage_trainer/data.py (hunk 3 above)
Comment (theRealSuperMario) [reply examples/multistage_trainer/data.py:1]: fixed

**Comment by theRealSuperMario** [comment]
can we morge thiis now?

**Review by mritv — APPROVED** [review]

**Pull request merged**